commit 939735a392169a5ac814c2be0cb41a6c14c9bb29
Author:     Yuhong Wen <yuhongw@nvidia.com>
AuthorDate: Tue Jan 7 09:28:44 2025 -0500
Commit:     GitHub <noreply@github.com>
CommitDate: Tue Jan 7 09:28:44 2025 -0500

    Add the fed event runner to the SP and CP. (#3129)
---
 nvflare/private/fed/client/client_engine.py | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/nvflare/private/fed/client/client_engine.py b/nvflare/private/fed/client/client_engine.py
index 4799ec91..a3417e28 100644
--- a/nvflare/private/fed/client/client_engine.py
+++ b/nvflare/private/fed/client/client_engine.py
@@ -39,6 +39,7 @@ from nvflare.private.fed.utils.app_deployer import AppDeployer
 from nvflare.private.fed.utils.fed_utils import security_close
 from nvflare.private.stream_runner import ObjectStreamer
 from nvflare.security.logging import secure_format_exception, secure_log_traceback
+from nvflare.widgets.fed_event import ClientFedEventRunner
 
 from .client_engine_internal_spec import ClientEngineInternalSpec
 from .client_executor import JobExecutor
@@ -99,6 +100,8 @@ class ClientEngine(ClientEngineInternalSpec, StreamableEngine):
         self.logger = get_obj_logger(self)
         self.fl_components = [x for x in self.client.components.values() if isinstance(x, FLComponent)]
 
+        self.fl_components.append(ClientFedEventRunner())
+
     def fire_event(self, event_type: str, fl_ctx: FLContext):
         fire_event(event=event_type, handlers=self.fl_components, ctx=fl_ctx)
 
@@ -470,6 +473,13 @@ class ClientEngine(ClientEngineInternalSpec, StreamableEngine):
     def get_all_job_ids(self):
         return self.client_executor.get_run_processes_keys()
 
+    def fire_and_forget_aux_request(
+        self, topic: str, request: Shareable, fl_ctx: FLContext, optional=False, secure=False
+    ) -> dict:
+        return self.send_aux_request(
+            topic=topic, request=request, timeout=0.0, fl_ctx=fl_ctx, optional=optional, secure=secure
+        )
+
 
 def shutdown_client(federated_client, touch_file):
     with open(touch_file, "a"):
